contributors/design-proposals/custom-metrics-api.md

```diff
@@ -0,0 +1,302 @@
+discovery document can be published by the adapter's API server, allowing
+clients to discover the available metrics.
+
+Note that we introduce the syntax of having a name of ` * ` here since
+there is no current syntax for getting the output of a subresource on
+multiple objects.
+
+API Objects
+-----------
+
+The request URLs listed above will return either the `MetricValueList` or
+`MetricValue` types (depending on whether they describe multiple objects,
+or a single one), described below:
+
+```go
+
+// a list of values for a given metric for some set of objects
+type MetricValueList struct {
+    unversioned.TypeMeta `json:",inline"`
+    unversioned.ListMeta `json:"metadata,omitempty"`
+
+    // the value of the metric across the described objects
+    Items []MetricValue `json:"metricValues"`
+}
+
+// a metric value for some object
+type MetricValue struct {
+    // a reference to the described object
+    DescribedObject ObjectReference `json:"describedObject"`
+
+    // the name of the metric
+    MetricName string `json:"metricName"`
+
+    // indicates the time at which the metrics were produced
+    Timestamp unversioned.Time `json:"timestamp"`
+
+    // indicates the window ([Timestamp-Window, Timestamp]) from
+    // which these metrics were calculated, when returning rate
+    // metrics calculated from cumulative metrics (or zero for
+    // non-calculated instantaneous metrics).
+    Window  unversioned.Duration `json:"window"`
+
+    // the value of the metric for this
+    Value resource.Quantity
+}
+```
+
+For instance, the example request above would yield the following object:
+
+```json
+{
+    "kind": "MetricValueList",
+    "apiVersion": "custom-metrics/v1alpha1",
+    "items": [
+        {
+            "metricName": "hits-per-second",
+            "describedObject": {
+                "kind": "Ingress",
+                "apiVersion": "extensions",
+                "name": "server1",
+                "namespace": "webapp"
+            },
+            "timestamp": SOME_TIMESTAMP_HERE,
+            "window": "10s",
+            "value": "10"
+        },
+        {
+            "metricName": "hits-per-second",
+            "describedObject": {
+                "kind": "Ingress",
+                "apiVersion": "extensions",
+                "name": "server2",
+                "namespace": "webapp"
+            },
+            "timestamp": ANOTHER_TIMESTAMP_HERE,
+            "window": "10s",
+            "value": "15"
+        }
+    ]
+}
+```
+
+Semantics
+---------
+
+### Object Types ###
+
+In order to properly identify resources, we must use resource names
+qualified with group names (since the group for the requests will always
+be `custom-metrics`).
+
+The `object-type` parameter should be the string form of
+`unversioned.GroupResource`.  Note that we do not include version in this;
+we simply wish to uniquely identify all the different types of objects in
+Kubernetes.
```

Comment:
a couple examples here would be helpful (maybe for pods in the legacy group, and one for jobs in the batch group)